Automattic/wp-calypso
# Plans overhaul: Add-ons page/route in my-sites

#### Proposed Changes

Adds the add-ons management page in Calypso.

Addresses 760-gh-Automattic/martech
Requires D82535-code

#### Media

<img width="700" alt="Screenshot 2022-06-08 at 5 11 04 PM" src="https://user-images.githubusercontent.com/1705499/172638415-6103a6cf-285f-4b21-b411-55125cd94ddf.png">

<img width="700" alt="Screenshot 2022-06-08 at 5 06 30 PM" src="https://user-images.githubusercontent.com/1705499/172638446-60aa4cb8-9631-4bbb-85dd-1e646eab4ae7.png">


#### Testing Instructions



* Apply D82535-code (and enable constant in `wp-content/mu-plugins/0-sandbox.php`)
* Go to `/add-ons/[SITE_URL]` to test the new section.
* New section is under "Upgrades"
* Clicking on "Buy add-on" should direct to checkout.
* Once purchased, clicking on "Manage add-on" should direct to purchases.

#### TODO

    - [ ] Investigate translations with API-fetched metadata (add-on name, description, term, currency, etc.). 
    - [x] Ensure switching locales properly refreshes the data.
- [x] Get some feedback on design and content:
    - [x] Buttons default to Calypso, probably best to leave that way here (not override to blue).
    - [x] Billing info just pulls whatever is returned from API.
    - [ ] Media query/breakpoint for the grid may need revising.
    - [ ] Checkmark, other styles, text, etc.
- [x] Address 762-gh-Automattic/martech to add the new section from the API.
- [x] Decide on alignment with site plan:
    - ~whether to show the section/page only for free sites~
    - or mark add-ons as selected ("_✔️ Included in you plan_") that are included in site plan 
- [x] Mark "no-ads" add-on as selected if site on a paid plan
- [ ] Placeholders while data is loading/fetched from API (the page has dependencies on `purchases` and `products`, and it renders empty until they are fetched).
- [ ] Decide on "Buy all add-ons" button (might be best to leave for later, until storage upgrades are decided)
- [x] Decide on shipping (maybe experimentally).
- [x] Some cleanup, address failing tests.



Fixes Automattic/martech#760
